platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -1127,18 +1477,115 @@ namespace platoon_strategic_ihp
+                return MobilityRequestResponse::NO_RESPONSE;
+            }
+            
+            // UCLA: For "CUT_IN_FROM_SIDE" message, the strategy params is : "SIZE:xx,SPEED:xx,DTD:xx" ---> NEED TO ADD IN HEADER FILE
+            std::vector<std::string> inputsParams;
+            boost::algorithm::split(inputsParams, params, boost::is_any_of(","));
+
+            std::vector<std::string> applicantSize_parsed;
+            boost::algorithm::split(applicantSize_parsed, inputsParams[0], boost::is_any_of(":"));
+            int applicantSize = std::stoi(applicantSize_parsed[1]);
+            ROS_DEBUG_STREAM("applicantSize: " << applicantSize);
+
+            std::vector<std::string> applicantCurrentSpeed_parsed;
+            boost::algorithm::split(applicantCurrentSpeed_parsed, inputsParams[1], boost::is_any_of(":"));
+            double applicantCurrentSpeed = std::stod(applicantCurrentSpeed_parsed[1]);
+            ROS_DEBUG_STREAM("applicantCurrentSpeed: " << applicantCurrentSpeed);
+
+            // Calculate downtrack (m) base on ecef. 
+            lanelet::BasicPoint2d incoming_pose = ecef_to_map_point(msg.location);
+            double applicantCurrentDtd = wm_->routeTrackPos(incoming_pose).downtrack;
+            ROS_DEBUG_STREAM("applicantCurrentmemberUpdates from ecef pose: " << applicantCurrentDtd);
+
+            // UCLA: Use crosstrack to determine condition to check if joining vehicle is close by
+            double applicant_crosstrack = wm_->routeTrackPos(incoming_pose).crosstrack;
+            ROS_DEBUG_STREAM("applicant_crosstrack from ecef pose: " << applicant_crosstrack);
+            double two_lane_cross_error = 2*config_.maxCrosstrackError + 3.5;
+
+            // cut-in vehicle and platoon are in adjacent lanes.
+            bool isLateralNearBy = (abs(applicant_crosstrack - current_crosstrack_) < two_lane_cross_error);
+            ROS_DEBUG_STREAM("isLateralNearBy = " << isLateralNearBy);
+
+            // Check if we have enough room for that applicant
+            int currentPlatoonSize = pm_.getTotalPlatooningSize();
+            bool hasEnoughRoomInPlatoon = applicantSize + currentPlatoonSize <= maxPlatoonSize_;
+
+            // -- core condition to decided accept joiner or not
+            if (hasEnoughRoomInPlatoon && isLateralNearBy) 
+            {
+                ROS_DEBUG_STREAM("The current platoon has enough room for the applicant with size " << applicantSize);
+
+                // UCLA: change to read platoon front info
+                double currentFrontDtd = pm_.getPlatoonFrontDowntrackDistance();
+                double currentRearDtd = pm_.getPlatoonRearDowntrackDistance();
+                ROS_DEBUG_STREAM("The target cut-in platoon leader downtrack is " << currentFrontDtd << " m,  and the platoon rear downtrack is " << currentRearDtd << "m");
+                
+                // UCLA: check if cut-in vehicle is too far away from the platoon logitudinally
```

Comment:
If you determine isLateralNearby by using isVehicleNextLane(), then a value of true guarantees that host is close enough, so this whole check is not necessary here.